### PR TITLE
B 2458 add create program panel

### PR DESCRIPTION
**Pull request by alexbeno**

### CODE REVIEW

File: app/financials/[sponsorId]/_features/create-program-panel/create-program-panel.context.tsx [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+export const CreateProgramPanelContext = createContext<CreateProgramPanelContextInterface>({
+  open: () => {},
+  close: () => {},
+  back: () => {},
+  isOpen: false,
+  name: "",
+});
+
+export function CreateProgramPanelProvider({ children, sponsorId }: CreateProgramPanelContextProps) {
+  const { Panel, ...rest } = useSidePanel({ name: "create-progam" });
```
Comment (haydencleary) [inline app/financials/[sponsorId]/_features/create-program-panel/create-program-panel.context.tsx:22]:
```suggestion
  const { Panel, ...rest } = useSidePanel({ name: "create-program" });
```

File: design-system/molecules/image-input/image-input.types.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+interface ClassNames {
+  base: string;
+}
+
+export interface ImageInputPort<C extends ElementType> extends Partial<Variants>, FieldContainerPort {
+  as?: C;
+  htmlProps?: ComponentPropsWithoutRef<C>;
+  classNames?: Partial<ClassNames>;
+  name: string;
+  max_size_mo?: number;
```
Comment (haydencleary) [inline design-system/molecules/image-input/image-input.types.ts:17]: camel case ?

File: shared/features/side-panels/side-panel/side-panel.tsx [hunk 3]
```diff
@@ -122,17 +135,39 @@ export const useSidePanel = (
   return useMemo(() => {
     return {
       Panel,
-      open: () => ref.current?.open(config),
-      close: current => ref.current?.close(current),
+      open: (data?: T) => ref.current?.open(data, config),
+      close: (current?: boolean) => ref.current?.close(current),
```
Comment (haydencleary) [inline shared/features/side-panels/side-panel/side-panel.tsx:139]: Is boolean the right type here?

File: shared/panels/allocate-program-sidepanel/allocate-program-sidepanel.types.ts [hunk 4]
```diff
@@ -1,4 +1 @@
-export interface AllocateProgramSidepanelProps {
-  sponsorId: string;
-  programId?: string;
-}
+export interface AllocateProgramSidepanelProps {}
```
Comment (haydencleary) [inline shared/panels/allocate-program-sidepanel/allocate-program-sidepanel.types.ts:1]: Delete the file

File: app/financials/[sponsorId]/_features/create-program-panel/create-program-panel.tsx [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+    };
+
+    await createProgram(createProgramData);
+
+    close();
+  }
+
+  return (
+    <>
+      <form onSubmit={handleSubmit(onCreateProgram)} className={"flex h-full w-full flex-col gap-px"}>
```
[the remark this answers is not in the document]
Comment (haydencleary) [reply app/financials/[sponsorId]/_features/create-program-panel/create-program-panel.tsx:58]: Maybe give the side panel component an `as` prop ?

File: shared/features/side-panels/side-panel/side-panel.tsx (hunk 3 above)
Comment (alexbeno) [reply shared/features/side-panels/side-panel/side-panel.tsx:139]: Capture d’écran 2024-09-11 à 12 08 24

yop

**Pull request merged**